validation/validator.go

```diff
@@ -67,3 +84,130 @@ func (v *Validator) Errors() httpvalidate.Errors {
 func (v *Validator) Fails() bool {
 	return v.instance.IsFail()
 }
+
+func getFieldKey(structField reflect.StructField) string {
+	if formTag := structField.Tag.Get("form"); formTag != "" {
+		return formTag
+	}
+	if jsonTag := structField.Tag.Get("json"); jsonTag != "" {
+		return jsonTag
+	}
+	return structField.Name
+}
+
+func setFieldValue(field reflect.Value, value any) error {
+	castedValue, err := castValueToType(field, value)
+	if err != nil {
+		return fmt.Errorf("%s %s: %w", errCastValueField, field.Type().String(), err)
+	}
+
+	field.Set(castedValue)
+	return nil
+}
+
+func createTagToFieldMap(val reflect.Value) map[string]reflect.Value {
+	tagToFieldMap := make(map[string]reflect.Value)
+	typ := val.Type()
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Field(i)
+		if !field.CanSet() {
+			continue
+		}
+		structField := typ.Field(i)
+		tag := getFieldKey(structField)
+		tagToFieldMap[tag] = field
+		tagToFieldMap[strings.ToLower(tag)] = field
+		tagToFieldMap[str.Camel2Case(tag)] = field
```

Comment:
Add two different cast to match common situations, like `UserName` `username` and `user_name`.